File: Factorial.cpp

#include <iostream>
#include <conio>
using namespace std;
int main() {
   int n , fact = 1, i;
   cout<<"Enter a postive number"<<endl;
   cin>>n;
   for(i=1; i<=n; i++)
   fact = fact * i;
   cout<<"Factorial of "<<n<<" is "<<fact;
   return 0;
}

File: factorial.java
package Maths;

public class factorial {
    static long fact(int n)
    {
        if(n==0)
        {
            return 1;
        }
        return n*fact(n-1);
    }
    public static void main(String[] args) {

        System.out.println(fact(20));

    }
}
